make.js

```diff
@@ -98,6 +99,37 @@ target.clean = function () {
     rm('-Rf', path.join(__dirname, '_test'));
 };
 
+//
+// Generate documentation (currently only YAML snippets)
+// ex: node make.js gendocs
+// ex: node make.js gendocs --task ShellScript
+//
+target.gendocs = function() {
+
+    var docsDir = path.join(__dirname, '_gendocs');
+    rm('-Rf', docsDir);
+    mkdir('-p', docsDir);
+    console.log();
+    console.log('> generating docs');
+
+    taskList.forEach(function(taskName) {
+        var taskPath = path.join(__dirname, 'Tasks', taskName);
+        ensureExists(taskPath);
+
+        // load the task.json
+        var taskJsonPath = path.join(taskPath, 'task.json');
+        if (test('-f', taskJsonPath)) {
+            var taskDef = require(taskJsonPath);
```

Comment:
it's better to `JSON.parse(fs.readFileSync(taskJsonPath).toString())`

require caches and it's very confusing to debug. I have gotten myself into that confusing situation a few times already and have been trying to switch the build scripts in this repo over whenever I touch an area.